### ISSUE TITLE
Pandas DataFrame/Series compatibility issue with abstract and functional API

### ISSUE THREAD

**Issue by hnazkani** [mrjbq7/ta-lib#215]
Hi,
I am trying to run a very basic program to calculate SMA from pandas.Series and getting error:
TypeError: only length-1 arrays can be converted to Python scalars 

Code to regenerate the issue pasted below:

import numpy as np, pandas as pd
import talib.abstract as ta

arr = np.arange(80).reshape(16,5)
df = pd.DataFrame(arr, columns=('open','high','low', 'close', 'volume'))
df = df.astype('float')
print ( ta.SMA(df['close'], timeperiod=3) )

**Comment by djporterco** [comment]
I was having the same issue,

I just added this section and everything worked as expected.

`inputs = {
    'open': np.random.random(100),
    'high': np.random.random(100),
    'low': np.random.random(100),
    'close': np.random.random(100),
    'volume': np.random.random(100)
}`
I replaced the `np.random.random(100)` with df['column names']..

hope this helps

**Comment by hnazkani** [comment]
@djporterco rterco Thanks for the reply. But my simple use case is that abstract API functions are not working with Pandas Series. Rephrasing this differently, below is another simpler example:

import numpy as np, pandas as pd, talib.abstract as ta
s = pd.Series(np.random.random(50))
print ( ta.SMA( s, timeperiod=3 ) )

Above code fails with the same error. Surprisingly using ndarray also doesn't work (using s.values instead of s also fails). 
I just want to pass a plain 1D array or pandas.Series (named 1D array) to abstract API's function and I am not sure why I should be required to specify the full 'open', 'close', 'volume' indicators for that.
Surprisingly, if we use "import talib as ta" then above code works perfectly. 

In a nutshell, pandas.Series is working fine with function API but pandas.DataFrame is not working at all with function APIs. Similarly pandas.DataFrame is working fine with abstract APIs but pandas.Series is not working Abstract APIs. So technically, i can't use the talib for any practical purpose, unless i write my own wrapper over and above this.

To me it looks like a bug or an inconsistent design issue between abstract and function API (ideally there should be only 1 with full functionalities exposed).

**Comment by mrjbq7** [comment]
It does look like some kind of bug using the ``pandas.DataFrame`` and the Abstract API.

This might be related to a change ``pandas`` did a long time ago where ``pandas.Series`` stopped subclassing ``numpy.array``.  Perhaps we didn't notice it stopped working in the Abstract API.

**Comment by hnazkani** [comment]
@mrjbq7 ideally it would be good to expose only one API to the external users and spend development efforts on the same. I am sure many pandas users would have tried TaLib and faced this issue in the past but would have gotten confused between the functional vs abstract and gone ahead with their own wrapper. 
TaLib is a terrific library and it would be highly desirable to have close integration with Pandas which is the de-facto standard in data science/financial time series analysis.

**Comment by mrjbq7** [comment]
TA-Lib itself actually exposes two separate APIs, but I agree it would be nice to keep it simple, and we clearly need to fix some pandas integration issue with the Abstract API. @briancappello ?
